source/dagger/gameplay/common/simple_collisions.h

```diff
@@ -2,15 +2,26 @@
 #include "core/system.h"
 #include "core/core.h"
 
+
 using namespace dagger;
 
+enum class EHitbox {
+    Rectangular = 1,
+    Circular = 2,
+};
+
 struct SimpleCollision
 {
-    Vector2 size;
+    Vector2 size; // If the hitbox is rectangular, size represents height and width, if circular, diameter and angle of coverage (currently unused)
```

Comment:
you can use c++ union here maybe.

Reply:
it'll make code more readable

Reply:
It crossed my mind at some point that a union could be used, but I just wanted to make code functional at first.